src/main/java/frc/robot/subsystems/drivetrain/Drivetrain.java

```diff
@@ -0,0 +1,167 @@
+    leftMiddleTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_LEFT_MIDDLE_TALON);
+    leftFrontTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_LEFT_FRONT_TALON);
+    rightRearTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_RIGHT_REAR_TALON);
+    rightMiddleTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_RIGHT_MIDDLE_TALON);
+    rightFrontTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_RIGHT_FRONT_TALON);
+    drivetrain = new DifferentialDrive(new SpeedControllerGroup(leftRearTalon, leftMiddleTalon, leftFrontTalon),
+        new SpeedControllerGroup(rightRearTalon, rightMiddleTalon, rightFrontTalon));
+    // TODO: set correct port for pigeon gyro.
+    gyro = new Pigeon(robotConstants.can.DRIVETRAIN_LEFT_REAR_TALON);
+    kinematics = new DifferentialDriveKinematics(robotConstants.drivetrainConstants.WHEEL_BASE_WIDTH);
+    odometry = new DifferentialDriveOdometry(Rotation2d.fromDegrees(getAngle()));
+    //TODO: set correct talons for encoders.
+    leftEncoder = new WPI_TalonSRX(robotConstants.can.TEMPORARY_TALON_FOR_LEFT_DRIVETRAIN_ENCODER);
+    rightEncoder = new WPI_TalonSRX(robotConstants.can.TEMPORARY_TALON_FOR_RIGHT_DRIVETRAIN_ENCODER);
+  }
+
+  // Drive functions
+  public void arcadeDrive(double x, double y) {
+    drivetrain.arcadeDrive(x, y);
+  }
+
+  public void curvatureDrive(double x, double y, boolean quickTurn) {
+    drivetrain.curvatureDrive(x, y, quickTurn);
+  }
+
+  public void tankDrive(double leftSpeed, double rightSpeed) {
+    drivetrain.tankDrive(leftSpeed, rightSpeed);
+  }
+
+  public void voltageTankDrive(double left, double right) {
+    tankDrive(left / RobotController.getBatteryVoltage(), right / RobotController.getBatteryVoltage());
+  }
+
+  // Gyro functions
+  double getAngle() {
+    return gyro.getAngle();
+  }
+
+  public double getRadianAngle() {
+    return Math.toRadians(getAngle());
+  }
+
+  public void resetGyro() {
+    gyro.reset();
+  }
+
+  public void calibrateGyro() {
+    gyro.calibrate();
+  }
+
+  // Encoders functions
+  // TODO: set correct talons that have encoder connected to it.
+
+  public int getLeftTicks() {
+    return leftEncoder.getSelectedSensorPosition();
+  }
+
+  public int getRightTicks() {
+    return rightEncoder.getSelectedSensorPosition();
+  }
+
+  public void resetEncoders() {
+    leftEncoder.setSelectedSensorPosition(0);
+    rightEncoder.setSelectedSensorPosition(0);
+  }
+
+  public double getLeftDistance() {
+    return getLeftTicks() / robotConstants.drivetrainConstants.LEFT_ENCODER_TICKS_PER_METER;
+  }
+  
+  public double getRightDistance() {
+    return getRightTicks() / robotConstants.drivetrainConstants.RIGHT_ENCODER_TICKS_PER_METER;
+  }
+
+  public double getAverageDistance() {
+    return (getLeftDistance() + getRightDistance()) / 2;
+  }
+
+  public double getRightVelocity() {
+    return rightEncoder.getSelectedSensorVelocity()
+        / robotConstants.drivetrainConstants.RIGHT_ENCODER_TICKS_PER_METER;
```

Comment:
leftEncoder.getSelectedSensorVelocity() returns units per 100 miliseconds (0.1 seconds). multply velocity by 10 in order to get meter/second